server/src/main/java/org/opensearch/gateway/remote/ClusterMetadataManifest.java

```diff
@@ -245,6 +258,20 @@ private static ClusterStateDiffManifest diffManifest(Object[] fields) {
         declareParser(PARSER_V3, CODEC_V3);
     }
 
+    public static int getCodecForVersion(Version version) {
```

Comment:
would it be simple to generate a map from the CODEC_TO_VERSION_MAPPING for every version up to the current version. This way we dont have to run a for loop for every manifest.

Reply:
you can auto generate the map, it need not be hard coded.